plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -129,13 +145,42 @@ public void storeMergedRows(Page page)
 
         mergePage.getInsertionsPage().ifPresent(insertPageSink::appendPage);
 
+        if (isCDFEnabled && mergePage.getInsertionsPage().isPresent()) {
+            if (cdfPageSink == null) {
+                cdfPageSink = cdfPageSinkSupplier.get();
+            }
+            Page insertionPage = mergePage.getInsertionsPage().get();
+            Block[] cdfPostUpdateBlocks = new Block[nonSynthesizedColumns.size() + 1];
+            for (int i = 0; i < nonSynthesizedColumns.size(); i++) {
+                cdfPostUpdateBlocks[i] = insertionPage.getBlock(i);
+            }
+            cdfPostUpdateBlocks[nonSynthesizedColumns.size()] = RunLengthEncodedBlock.create(
```

Comment:
but that would be incorrect, `insertionPage` is smaller than size of this array because `cdfPostUpdateBlocks` contains all blocks from `insertionPage` + `name of the operation` like `update_postimage`

Reply:
Sorry, had the wrong variable name. I should have said `cdfPostUpdateBlocks[cdfPostUpdateBlocks.size - 1]`

The RLE is always the last element in the Block array, right?